openbb_terminal/econometrics/econometrics_view.py

```diff
@@ -368,10 +368,7 @@ def display_cointegration_test(
     """
 
     if isinstance(datasets, pd.DataFrame):
-        new_datasets = {}
-        for column in datasets.columns:
-            new_datasets[column] = datasets[column]
-
+        new_datasets = {column: datasets[column] for column in datasets.columns}
```

Comment:
Regarding this:
Why is this here at all? This causes the flake8 lint to fail

Reply:
The variable `new_datasets` is not being used later on after the if-test

Reply:
@DidierRLopes  Do you know?
I'm inclined towards moving the unused variable. There seems to be no usage of it in the code.

Reply:
This was done by @JerBouma, probably he can shed some insight here

Reply:
I think a lot of these will come forward as I open PR's for the different refactors btw. 😅

Reply:
@JerBouma  Could you shed some insight regarding the usage of the unused new_datasets variable?

Reply:
Seems to be legacy code as it is indeed now unused. There have been multiple refractors done to the code in which this got optimized apparently.

Reply:
Thanks a lot @JerBouma!
@DidierRLopes Perhaps, going forward, if a variable is unused, we remove it? Unless anything else is explicitly mentioned on the code?

Reply:
@Uzaaft yes, definitely unused variables should not be in the code. I wonder how that one passed the linters.

Would you be up for jumping into a call on Discord or elsewhere to coordinate the contributing efforts? This would make it easier and quicker to merge PRs

Reply:
Yeah sure thing @piiq.  
After the merging that you did, I saw a few more possibilities of using list-comprehensions, so I am including those as well now.

Reply:
@Uzaaft can you add me on discord please? `pll_llq#8920`

Reply:
I just sent you an request @piiq